ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/apps/[externalID]/syncs/Sync/FunctionList.tsx

```diff
@@ -31,57 +49,111 @@ export function FunctionList({ className, removedFunctions, syncedFunctions }: P
-        <CheckCircleIcon className="mr-1 text-green-600" height={20} />
-        <h2>Synced Functions ({syncedFunctions.length})</h2>
-      </div>
-      <div className="flex border-b border-slate-300 p-2">
-        <MinusCircleIcon className="mr-1 text-red-600" height={20} />
-        <h2>Removed Functions ({removedFunctions.length})</h2>
-      </div>
+    <>
+      <CollapsibleCardRoot
+        type="single"
+        defaultValue={openSyncedFunctions ? 'syncedFunctions' : undefined}
+        collapsible
+      >
+        <CollapsibleCardItem value="syncedFunctions">
+          <CollapsibleCardHeader className="flex items-center justify-between border-slate-300 px-6 py-3 text-sm font-medium text-slate-600 data-[state=open]:border-b">
+            <h2>Synced Functions ({syncedFunctions.length})</h2>
+            <CollapsibleCardTrigger
+              asChild
+              onClick={() => setSyncedFunctions(!openSyncedFunctions)}
+            >
+              <Button
+                className="group"
+                appearance="outlined"
+                icon={
+                  <ChevronDownIcon className="transform-90 text-slate-500 transition-transform duration-500 group-data-[state=open]:-rotate-180" />
+                }
+              />
+            </CollapsibleCardTrigger>
+          </CollapsibleCardHeader>
+          <CollapsibleCardContentWrapper>
+            {openSyncedFunctions && (
+              <CollapsibleCardContent>
+                {syncedFunctions.map((fn, i) => {
+                  const isLast = i === syncedFunctions.length - 1;
 
-      <div className="border-r border-slate-300">
-        {syncedFunctions.map((fn, i) => {
-          const isLast = i === syncedFunctions.length - 1;
+                  return (
+                    <Link
+                      href={`/env/${env.slug}/functions/${encodeURIComponent(fn.slug)}`}
+                      key={fn.id}
+                    >
+                      <div
+                        className={classNames(
+                          'group flex w-full items-center gap-2 border-slate-200 py-3 pl-6 pr-2 text-sm font-medium text-slate-700 hover:bg-indigo-50  hover:text-indigo-600',
+                          !isLast && 'border-b'
+                        )}
+                      >
+                        {fn.name}
+                        <ArrowRightIcon className="h-3 w-3 -translate-x-3 text-indigo-600 opacity-0 transition-all group-hover:translate-x-0 group-hover:opacity-100" />
+                      </div>
+                    </Link>
+                  );
+                })}
 
-          return (
-            <Link href={`/env/${env.slug}/functions/${encodeURIComponent(fn.slug)}`} key={fn.id}>
-              <div
-                className={classNames(
-                  'border-slate-200 p-2 hover:bg-indigo-50',
-                  !isLast && 'border-b'
+                {syncedFunctions.length === 0 && (
```

Comment:
This should be an impossible state (an app can't have 0 functions). Maybe we should throw an error when it happens so that it shows in Sentry